surface/include/pcl/surface/impl/marching_cubes.hpp

```diff
@@ -288,8 +273,28 @@ pcl::MarchingCubes<PointNT>::performReconstruction (pcl::PointCloud<PointNT> &po
     return;
   }
 
+  // real part of marching cube
+  performReconstructionProc();
+
+  points.swap(intermediate_cloud_);
+
+  polygons.resize (points.size () / 3);
+  for (size_t i = 0; i < polygons.size (); ++i)
+  {
+    pcl::Vertices v;
+    v.vertices.resize (3);
+    for (int j = 0; j < 3; ++j)
+      v.vertices[j] = static_cast<int> (i) * 3 + j;
+    polygons[i] = v;
+  }
+}
```

Comment:
This section can also be refactored into `performReconstructionProc`. It will required you to change the signature of that method to include the pointcloud and the polygon vector.